[TF-TRT]  Refactor Softmax converter to use OpConverterBase

pavanimajety opened the pull request:
This commit refactors the Softmax TF-TRT conversion to use OpConverterBase classes.

bixia1 commented:
The dependent PR is merged, can you rebase?

bixia1 commented on tensorflow/compiler/tf2tensorrt/convert/ops/softmax.cc:
```diff
@@ -0,0 +1,67 @@
+#if GOOGLE_CUDA && GOOGLE_TENSORRT
```
Please add license header, can copy and paste from another source file then change the year to 2022.

bixia1 commented on tensorflow/compiler/tf2tensorrt/convert/ops/softmax.cc:
```diff
@@ -0,0 +1,67 @@
+        params.converter->network()->addSoftMax(*logits_tensor->trt_tensor());
+    TFTRT_RETURN_ERROR_IF_NULLPTR(layer, node_def.name());
+    params.converter->SetLayerName(layer, node_def);
+    // Tensorflow SoftMax assumes applying softmax on the last dimension.
```
s/assumes applying/applies/

bixia1 reviewed (changes requested):
compilation errors:
[third_party/tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc:105](https://cs.corp.google.com/piper///depot/google3/third_party/tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc?l=105&ws=tap-prod-presubmit/207445590&snapshot=2):11: error: enumeration value 'kSOFTMAX' not handled in switch [-Werror,-Wswitch]
  switch (layer_type) {
          ^~~~~~~~~~
1 error generated.

[third_party/tensorflow/compiler/tf2tensorrt/convert/ops/softmax.cc:43](https://cs.corp.google.com/piper///depot/google3/third_party/tensorflow/compiler/tf2tensorrt/convert/ops/softmax.cc?l=43&ws=tap-prod-presubmit/207445590&snapshot=2):17: error: unused variable 'node_def' [-Werror,-Wunused-variable]
    const auto &node_def = params.node_def;
                ^
1 error generated.

gbaned commented:
Hi @pavanimajety Can you please resolve conflicts? Thank you!

bixia1 reviewed (commented):
Would you please squash the commits?

pavanimajety committed:
[TF-TRT]  Refactor Softmax converter to use OpConverterBase

Change softmax's TF-TRT converter to OpConverter class infrastructure.
Fix unused variable error, switch case compilation error, conflicts and remove whitespace
